src/pybind/mgr/nfs/export.py

```diff
@@ -382,10 +376,18 @@ def _delete_export(
                 export = self._fetch_export(cluster_id, pseudo_path)
 
             if export:
+                exports_count = 0
+                if export.fsal.name == NFS_GANESHA_SUPPORTED_FSALS[0]:
+                    exports_count = self.get_export_count_by_cmount_path(export.fsal.cmount_path,
+                                                                         cluster_id)
+                    if exports_count == 1:
+                        self._delete_export_user(export)
```

Comment:
can you elaborate on this please, so if there is an export which is cephfs' and if the num of exports is 'one' then we delete the export user? what about cases for more than one?

Reply:
There can be multiple exports sharing same cephfs client, so we need to make sure on deletion that if there are more than 1 export sharing that client, if that is we shouldn't felete the user else if its just one we remove it!

Reply:
So this means we delete the user if and only if there is a single export with the specified fs name and cmount path. So can there be an edge case possible where we're left with a lingering user?

Reply:
(thinking out loud) Let's say we have two exports `a` and `b` with a fs `foo` and cmount = `/` and user `user1`.  Deleting `b` skips deleting `user1`, now the export `a` is abruptly killed (some OOM happened lets say). Now, since there is no `a`, the delete export should not work. So can we say that the user `user1` can linger around or are there any fallback mechanisms in place? (yet to run through the entire PR, this is just something that camp up during an initial scramble.)

Reply:
> (thinking out loud) Let's say we have two exports `a` and `b` with a fs `foo`, cmount = `/` and user `user1`. Deleting `b` skips deleting `user1`, now the export `a` is abruptly killed (some OOM happened lets say). Now, since there is no `a`, the delete export should not work. So can we say that the user `user1` can linger around or are there any fallback mechanisms in place? (yet to run through the entire PR, this is just something that camp up during an initial scramble.)

@vshankar any thoughts?

Reply:
Had a conversation with @dparmar18 regarding this. There should be exactly the same number of calls to delete and export compared to creating an export -- the last export delete would remove the user. If the mounted export gets OOM killed or whatever, it isn't a problem for this plugin.

Reply:
Thanks @vshankar